Microsoft.Azure.CosmosRepository/src/ChangeFeed/Providers/DefaultChangeFeedContainerProcessorProvider.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.CosmosRepository.Builders;
+using Microsoft.Azure.CosmosRepository.Options;
+using Microsoft.Azure.CosmosRepository.Providers;
+using Microsoft.Azure.CosmosRepository.Services;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.CosmosRepository.ChangeFeed.Providers
+{
+    internal class DefaultChangeFeedContainerProcessorProvider : IChangeFeedContainerProcessorProvider
+    {
+        private readonly IOptionsMonitor<RepositoryOptions> _optionsMonitor;
+        private readonly ICosmosContainerService _containerService;
+        private readonly ILeaseContainerProvider _leaseContainerProvider;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly IChangeFeedOptionsProvider _changeFeedOptionsProvider;
+
+        public DefaultChangeFeedContainerProcessorProvider(IOptionsMonitor<RepositoryOptions> optionsMonitor,
+            ICosmosContainerService containerService,
+            ILeaseContainerProvider leaseContainerProvider,
+            ILoggerFactory loggerFactory,
+            IServiceProvider serviceProvider,
+            IChangeFeedOptionsProvider changeFeedOptionsProvider)
+        {
+            _optionsMonitor = optionsMonitor;
+            _containerService = containerService;
+            _leaseContainerProvider = leaseContainerProvider;
+            _loggerFactory = loggerFactory;
+            _serviceProvider = serviceProvider;
+            _changeFeedOptionsProvider = changeFeedOptionsProvider;
+        }
+
+        public IEnumerable<IContainerChangeFeedProcessor> GetProcessors()
+        {
+            RepositoryOptions repositoryOptions = _optionsMonitor.CurrentValue;
+
+            IEnumerable<IGrouping<string, ContainerOptionsBuilder>> containers = repositoryOptions.ContainerBuilder
+                .Options
+                .Where(x => x.ChangeFeedOptions != null)
+                .GroupBy(x => x.Name);
+
+            foreach (IGrouping<string, ContainerOptionsBuilder> container in containers)
+            {
+                List<Type> itemTypes = container.Select(x => x.Type).ToList();
+
+                yield return new DefaultContainerChangeFeedProcessor(
```

Comment:
@robertbennett1998 could this be tidied up a little?

Reply:
I was thinking rather than passing the services down, create the two containers here and pass them in? Makes this class more standalone and not tied to DI and injecting these providers